commands/proxy/proxy-set.ts

```diff
@@ -22,22 +23,30 @@ export class ProxySetCommand extends ProxyCommandBase {
 	}
 
 	public async execute(args: string[]): Promise<void> {
-		let hostname = args[0],
-			port = args[1],
-			username = args[2],
-			password = args[3];
+		let urlString = args[0],
+			username = args[1],
+			password = args[2];
+
+		const noUrl = !urlString;
+		if (noUrl) {
+			if (!isInteractive()) {
+				this.$errors.fail("Console is not interactive - you need to supply all command parameters.");
+			} else {
+				urlString = await this.$prompter.getString("Url", { allowEmpty: false });
+			}
+		}
 
-		const noHostName = !hostname;
+		const urlObj = parse(urlString);
+		let port = urlObj.port && +urlObj.port || HttpProtocolToPort[urlObj.protocol];
 		const noPort = !port || !this.isValidPort(port);
+		const authCredentials = this.getCredentialsFromAuth(urlObj.auth || "");
+		username = username || authCredentials.username;
```

Comment:
I believe we should fail in case you specify username in both ways, for example:
`appbuilder proxy set http://username:password@127.0.0.1:8888 username1 password1`
should fail. In case the username and password on both places are the same, we should not fail.